R/class-Megastudy.R

```diff
@@ -0,0 +1,97 @@
+check_study_vocabulary <- function(object) {
+  errors <- character()
+
+  if (is.na(object@studyIdColumnName)) {
+    errors <- c(errors, "StudyIdColName is required but not provided.")
+  }
+
+  if (is.na(object@variableSpec@variableId)) {
+    errors <- c(errors, "VariableSpec is required but not provided.") 
+  }
+
+  return(if (length(errors) == 0) TRUE else errors)  
+}
+
+#' @include class-VariableMetadata.R
+#' @export
+StudySpecificVocabulary <- setClass("StudySpecificVocabulary", representation(
+  studyIdColumnName = 'character',
```

Comment:
`studyIdColumnName` doesn't seem like it's about the vocabulary so much as the megastudy. Could this be a slot on the megastudy classs instead?

Reply:
its just so i know both the entityid.variableid of the study entity id 'variable' as well as its value/ study id for this particular study vocabulary.. importantly, all in one place